design-proposals/klip-42-schema-migrations-tool.md

```diff
@@ -399,18 +409,101 @@ are the basic configurations that the tool should support.
 Example of `ksql-migrations.properties`
 ```
 # Server URL and authentication
-ksql.server.url='http://localhost:8080'
-ksql.username='user1'
-ksql.password='pass1'
+ksql.server.url=http://localhost:8080
+ksql.username=user1
+ksql.password=pass1
+
+# TLS configs, including for mTLS
+ssl.truststore.location=/path/to/truststore
+ssl.truststore.password=pass
+ssl.keystore.location=/path/to/keystore
+ssl.keystore.password=pass
+ssl.key.password=pass
 
 # Migrations details
-ksql.migrations.streamName='migration_events'
-ksql.migrations.tableName='schema_version'
-ksql.migrations.createSchemas='true'
+ksql.migrations.stream.name=migration_events
+ksql.migrations.table.name=schema_version
+ksql.migrations.stream.topic.name=default_ksql_migration_events
+ksql.migrations.table.topic.name=default_ksql_schema_version
+ksql.migrations.topic.replicas=1
 ```
 
+The default stream and table metadata topic names will be `{clusterID}ksql_{StreamOrTableName}`, where the migrations
+tool with determine the ksqlDB cluster ID by accessing the `/info` endpoint of the ksqlDB server. If desired,
+these topic names may be overridden by the user.
+
 Note: The command line options and other configurations will also be defined during the implementation.
 
+### Migration state transitions and potential race conditions
+
+Below we have two potential options for how to manage migration state.
```

Comment:
Do people have thoughts between these two options? My preference is to start with the latter since it's simpler.

Reply:
I'd like to keep this simple too. I raised this race condition issue with @MichaelDrogalis during the KLIP review, and we thought we could keep it simple for the first version of the tool, and work on improving the race conditions in the future. Upgrades are usually run by one user or automated tool during an software upgrade process, so I don't expect many issues with this.

Reply:
Sounds good, I just wanted to confirm that we're OK with these possible race conditions. Sounds like we are so that's great.

Reply:
chiming in here as well, I'm OK with the potential race condition b/c of the justification that it's either run by a human or an automated tool. I don't expect a high throughput of users executing concurrent migrations (and it's not worth the ROI to implement something more complicated that would also make the minimum use requirements more difficult - e.g. I think you need to ensure your broker is set up with the correct configs to use transactions and you must have multiple brokers for it to "work" reliably)